src/cli/repl.rs

```diff
@@ -0,0 +1,442 @@
+                let current_env = &self.store.lurk_sym("current-env");
+                let binding = &self.store.list(&[first, second]);
+                let bindings = &self.store.list(&[*binding]);
+                let current_env_call = &self.store.list(&[*current_env]);
+                let expanded = &self.store.list(&[*l, *bindings, *current_env_call]);
+                let (expanded_io, ..) = self.eval_expr(*expanded)?;
+
+                self.env = expanded_io.expr;
+
+                let (new_binding, _) = &self.store.car_cdr(&expanded_io.expr)?;
+                let (new_name, _) = self.store.car_cdr(new_binding)?;
+                println!("{}", new_name.fmt_to_string(&self.store));
+            }
+            "defrec" => {
+                // Extends env with a recursive binding.
+                //
+                // This: !(:defrec foo (lambda () 123))
+                //
+                // Gets macroexpanded to this: (letrec ((foo (lambda () 123)))
+                //                               (current-env))
+                //
+                // And the state's env is set to the result.
+                let (first, second) = self.peek2(cmd, args)?;
+                let l = &self.store.lurk_sym("letrec");
+                let current_env = &self.store.lurk_sym("current-env");
+                let binding = &self.store.list(&[first, second]);
+                let bindings = &self.store.list(&[*binding]);
+                let current_env_call = &self.store.list(&[*current_env]);
+                let expanded = &self.store.list(&[*l, *bindings, *current_env_call]);
+                let (expanded_io, ..) = self.eval_expr(*expanded)?;
+
+                self.env = expanded_io.expr;
+
+                let (new_binding_outer, _) = &self.store.car_cdr(&expanded_io.expr)?;
+                let (new_binding_inner, _) = &self.store.car_cdr(new_binding_outer)?;
+                let (new_name, _) = self.store.car_cdr(new_binding_inner)?;
+                println!("{}", new_name.fmt_to_string(&self.store));
+            }
+            "load" => {
+                let first = self.peek1(cmd, args)?;
+                match self.store.fetch_string(&first) {
+                    Some(path) => {
+                        let joined = pwd_path.join(Path::new(&path));
+                        self.load_file(&joined)?
+                    }
+                    _ => bail!("Argument of `load` must be a string."),
+                }
+                io::Write::flush(&mut io::stdout()).unwrap();
+            }
+            "assert" => {
+                let first = self.peek1(cmd, args)?;
+                let (first_io, ..) = self.eval_expr(first)?;
+                if first_io.expr.is_nil() {
+                    bail!(
+                        "`assert` failed. {} evaluates to nil",
+                        first.fmt_to_string(&self.store)
+                    )
+                }
+            }
```

Comment:
This [fails to fail](https://www.youtube.com/watch?v=iuLBhxZUkmU&t=240s) with non-zero exit code:
```
Lurk REPL welcomes you.
> !(assert nil)
!Error: `assert` failed. nil evaluates to nil
> 
Exiting...
➜  lurk-rs git:(ap/clap-main) echo $?
0
```
For comparison, the current repl (although ugly) behaves as needed:
```
Lurk REPL welcomes you.
> !(assert nil)
thread 'main' panicked at 'assertion failed: !first_evaled.is_nil()', /Users/clwk/fil/lurk-rs/src/repl.rs:399:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
➜  lurk-rs ✗ echo $?
101
```

Reply:
Fixed, but I'm exiting with a custom code (`1`) instead of panicking